### PR TITLE
Batch Norm

### PR DESCRIPTION
**Pull request by feynmanliang**

### CODE REVIEW

File: flowtorch/bijectors/batch_norm.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        self.log_gamma = nn.Parameter(torch.zeros(num_inputs))
+        self.beta = nn.Parameter(torch.zeros(num_inputs))
+        self.register_buffer("running_mean", torch.zeros(num_inputs))
+        self.register_buffer("running_var", torch.ones(num_inputs))
+
+    def forward(self, x):
+        return (self.log_gamma, self.beta, self.running_mean, self.running_var)
+
+
+class BatchNormParams(flowtorch.Params):
```
Comment (feynmanliang) [inline flowtorch/bijectors/batch_norm.py:24]: Might be worthwhile adding some helper classes for these trivial hypernets which just register params/buffers in a `Module` and return them without transformation.

File: flowtorch/bijectors/batch_norm.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        return modules, None
+
+    def _forward(
+        self,
+        x: torch.Tensor,
+        context: Optional[torch.Tensor] = None,
+        modules: Optional[nn.ModuleList] = None,
+    ) -> Optional[Sequence[torch.Tensor]]:
+        assert isinstance(modules, flowtorch.ParamsModule)
```
Comment (feynmanliang) [inline flowtorch/bijectors/batch_norm.py:39]: These assertions / type casts are building up (see AffineAutoregressive as well); will `_forward` ever be called without `modules` (similarly, will `Bijector.forward` ever be called without `params`)?

**Comment by feynmanliang** [comment]
Looking at this again, `BatchNorm` seems to have a sense of direction (i.e. `forward`/`sample` vs `inverse`/`log_prob`) and this PR adds it for the `log_prob` case. If we are adopting the convention that `forward`/`sample` should be the fast path, then I think the current implementation is backwards?

**Comment by feynmanliang** [comment]
Will cherrypick off of `develop`
